src/public/js/util.js

```diff
@@ -38,6 +38,12 @@ var UTIL = (function() {
      0xb4, 0x7c, 0xb8, 0x26, 0x77, 0x99, 0xe3, 0xa5,
      0x67, 0x4a, 0xed, 0xde, 0xc5, 0x31, 0xfe, 0x18,
      0x0d, 0x63, 0x8c, 0x80, 0xc0, 0xf7, 0x70, 0x07]);
+  log3.get = function(x) {
```

Comment:
This is perhaps a bit more complicated of a talk then would be good for back and forth on here. Perhaps Facebook chat would be ok but probably next time we pseudo-talk I can explain better based on what background knowledge you do have in the areas (its easier to gage that in an actual discussion :P ) Block size is not easy to determine and not really the important part (In fact I have no idea what common block sizes are these days (for your curiosity though, I have seen 4KB as a frequent block size in the Qualcomm DSP I worked on once (but shhhhh))). The important part is more just recognizing that memory caches do work in blocks. The JS engine cache you are talking about however is something else entirely. It is still a cache since it does the same thing as a memory cache essentially, just in a different place. Browsers these days basically have 2 kinds of caches at the application layer plus they use the computers memory caches(there are multiple(unless you are using like a commodore 64 or something)). The first is the content cache. This cache basically saves a copy of webpages you visit and keeps them for a little while. This way if you go back to a website a little while after leaving it, instead of downloading the whole page from some server half way across the world, you simply ask the server if the page has changed since I last accessed it. If the server says no then you just look at the copy you have saved before. The other cache that browsers implement(and this tech is a bit newer I believe) is basically a compiled JS cache. What I mean by this is that when you go to a website that runs a JS script your browser downloads that JS script then compiles it then runs it. This compilation takes some time and uses some computing resources not to mention you also have to download the code. What if there is some JS library that is utilized on thousands of pages? Should you recompile that library every time you access any of those pages? This cache says no. It keeps a copy of the compiled output (plus a bunch of metadata presumably) of the JS scripts it sees frequently and then when it sees it again on a new page instead of downloading the script and compiling it, it just runs the saved compiled output on the new page being loaded as well.

So yeah cache is a commonly used computer word for a thing that holds data temporarily basically. How memory caches work and why the things I was saying are true is a more complicated discussion and this post already got super long so lets talk on it later whenever you have time. :)